### ISSUE TITLE
Trame ui can not hide menu

### ISSUE THREAD

**Issue by ConnorNoddin** [pyvista/pyvista#6464]
### Describe the bug, what's wrong, and what you expected.

ui() has a parameter add_menu=True. This works fine when it is true, however, when it is set to false there is a bug. The code always assigns view.menu to self.menu, even if menu does not exist. Consequently, the code breaks.

```python
            elif mode == "server":
                view = PyVistaRemoteView(self.plotter, **kwargs)
            elif mode == "client":
                view = PyVistaLocalView(self.plotter, **kwargs)

            self._html_views.add(view)
            if add_menu:
                view.menu = self.menu
        return view

```
This is all that is required to fix it.     

### Steps to reproduce the bug.

```python
self.view = plotter_ui(self.plotter, mode="trame", add_menu=False)
```

### System Information

```shell
--------------------------------------------------------------------------------
  Python 3.11.6 (main, Nov  2 2023, 14:08:07) [GCC 8.5.0 20210514 (Red Hat
  8.5.0-4)]

           pyvista : 0.43.3
               vtk : 9.3.0.dev0
             numpy : 1.26.4
        matplotlib : 3.8.3
            scooby : 0.9.2
             pooch : 1.8.1
            pillow : 10.2.0
           imageio : 2.34.1
           IPython : 8.25.0
             scipy : 1.12.0
              tqdm : 4.63.1
            meshio : 5.3.5
             trame : 3.6.3
      trame_client : 3.2.1
      trame_server : 3.0.3
         trame_vtk : 2.8.9
     trame_vuetify : 2.6.2
      nest_asyncio : 1.6.0
--------------------------------------------------------------------------------
```

**Comment by jourdain** [comment]
FYI @annehaley

**Comment by banesullivan** [comment]
Hey there, it looks like you are running on an older version of PyVista (0.43.3 posted above). This was resolved in #5668 and I just ran both of the following examples on the latest version to confirm.

Please upgrade PyVista to see this fix

```py
from trame.app import get_server
from trame.ui.vuetify3 import SinglePageLayout

import pyvista as pv
from pyvista import examples
from pyvista.trame.ui import get_viewer

pv.OFF_SCREEN = True

server = get_server(client_type='vue3')
state, ctrl = server.state, server.controller

mesh = examples.load_random_hills()

plotter = pv.Plotter()
actor = plotter.add_mesh(mesh, cmap='viridis')

viewer = get_viewer(plotter)

with SinglePageLayout(server) as layout:
    with layout.content:
        view = viewer.ui(add_menu=False, mode='trame')
        ctrl.view_update = view.update

server.start()
```


```py
from trame.app import get_server
from trame.ui.vuetify3 import SinglePageLayout

import pyvista as pv
from pyvista import examples
from pyvista.trame.ui import plotter_ui

pv.OFF_SCREEN = True

server = get_server(client_type='vue3')
state, ctrl = server.state, server.controller

mesh = examples.load_random_hills()

plotter = pv.Plotter()
actor = plotter.add_mesh(mesh, cmap='viridis')

with SinglePageLayout(server) as layout:
    with layout.content:
        view = plotter_ui(plotter, add_menu=False, mode='trame')
        ctrl.view_update = view.update

server.start()
```
